### PR TITLE
Remove string aliases for python2

### PR DESCRIPTION
**Pull request by Andrey-Raspopov**
This pull request addresses issue #2420 

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [ ] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Bio/Phylo/BaseTree.py [hunk 1]
```diff
@@ -152,7 +152,7 @@ def match(node):
             if not hasattr(node, key):
                 return False
             target = getattr(node, key)
-            if isinstance(pattern, basestring):
+            if isinstance(pattern, str):
                 return isinstance(target, basestring) and re.match(
```
Comment (peterjc) [inline Bio/Phylo/BaseTree.py]: Missed one, spotted by flake8 and me though.

**Comment by peterjc** [comment]
Two easily fixed issues spotted by flake8 run by TravisCI:

```
$ flake8
./Bio/Phylo/BaseTree.py:156:43: F821 undefined name 'basestring'
./Bio/Phylo/BaseTree.py:281:77: F821 undefined name 'unicode'
```

Anyway, the changes look good in principle - hopefully this won't cause merge conflicts with the competing efforts from other contributors.

File: Bio/Phylo/BaseTree.py (hunk 1 above)
Comment (Andrey-Raspopov) [reply Bio/Phylo/BaseTree.py]: I've changed my environment and missed flake on this one, yeah. I'll fix it in the next commit.

**Comment by Andrey-Raspopov** [comment]
Yeah, I saw BioSQL PR, but I suppose it should be okay, since I've started from the crutches, that are easiest to replace .

**Comment by peterjc** [comment]
Thank you - applied as a single commit via squash-and-merge because the conflicts prevented a rebase.

**Pull request merged**